client/my-sites/posts/post-controls/index.jsx

```diff
@@ -219,10 +214,43 @@ PostControls.propTypes = {
 	onToggleShare: PropTypes.func,
 	onTrash: PropTypes.func,
 	post: PropTypes.object.isRequired,
+	recordCopyPost: PropTypes.func,
+	recordEditPost: PropTypes.func,
+	recordPreviewPost: PropTypes.func,
+	recordViewPost: PropTypes.func,
+	recordViewPostStats: PropTypes.func,
 	site: PropTypes.object,
 	translate: PropTypes.func,
 };
 
-export default connect( ( state, { site, post } ) => ( {
+const mapStateToProps = ( state, { site, post } ) => ( {
+	canUserDeletePost: canCurrentUser( state, site.ID, 'delete_posts' ),
+	canUserEditPost: canCurrentUser( state, site.ID, 'edit_posts' ),
+	canUserPublishPost: canCurrentUser( state, site.ID, 'publish_posts' ),
 	isPublicizeEnabled: isPublicizeEnabled( state, site.ID, post.type ),
-} ) )( localize( PostControls ) );
+} );
+
+const mapDispatchToProps = dispatch => ( {
+	recordCopyPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Copy Post' ),
+		recordTracksEvent( 'calypso_post_controls_copy_post_click' ),
+	) ),
+	recordEditPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Edit Post' ),
+		recordTracksEvent( 'calypso_post_controls_edit_post_click' ),
+	) ),
+	recordPreviewPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Preview Post' ),
+		recordTracksEvent( 'calypso_post_controls_preview_post_click' ),
+	) ),
+	recordViewPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked View Post' ),
+		recordTracksEvent( 'calypso_post_controls_view_post_click' ),
+	) ),
+	recordViewPostStats: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked View Post Stats' ),
+		recordTracksEvent( 'calypso_post_controls_view_post_stats_click' ),
+	) ),
+} );
```

Comment:
sometimes it's just the way it is that the actual action we're talking about is specifically about tracking metrics. other times the metrics are happenstance riding along the back of a more meaningful action to the user: "Navigated to stats…" for example.

I see that the old version of this was simply calling analytics as the only action, but is there code that _does_ something on which we could compose `withAnalytics()`? just a question I like to ask.